common/src/main/java/net/buycraft/plugin/execution/ServerEventSenderTask.java

```diff
@@ -0,0 +1,61 @@
+package net.buycraft.plugin.execution;
+
+import com.google.common.collect.Lists;
+import net.buycraft.plugin.IBuycraftPlatform;
+import net.buycraft.plugin.data.ServerEvent;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Level;
+
+public class ServerEventSenderTask implements Runnable {
+
+    private final IBuycraftPlatform platform;
+    private final boolean verbose;
+
+    private final List<ServerEvent> eventQueue = Lists.newArrayList();
+
+    private boolean stopped = false;
+
+    public ServerEventSenderTask(IBuycraftPlatform platform, boolean verbose) {
+        this.platform = platform;
+        this.verbose = verbose;
+    }
+
+    @Override
+    public void run() {
+        if (platform.getApiClient() == null) {
+            return;
+        }
+
+        if(platform.getServerInformation() == null || !platform.getServerInformation().getAccount().isLogEvents()) {
+            return;
+        }
+
+        List<ServerEvent> runEvents = Lists.newArrayList(eventQueue);
+
+        try {
+            if (verbose) platform.log(Level.INFO, "Sending " + runEvents.size() + " analytic events");
+            platform.getApiClient().sendEvents(runEvents).execute();
```

Comment:
@theminecoder Can we please skip the HTTP request if the runEvents array is empty? Will cut down on un needed requests to the API.